### PR TITLE
[NBS Docs] Update adding basic permission tutorial

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Closes: https://github.com/backstage/backstage/issues/26334

Update the adding basic permission tutorial to use the new backend system.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/permissions/plugin-authors/02-adding-a-basic-permission-check--old.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
+---
+id: 02-adding-a-basic-permission-check
```
Comment (freben) [inline docs/permissions/plugin-authors/02-adding-a-basic-permission-check--old.md:2]:
```suggestion
id: 02-adding-a-basic-permission-check--old
```

File: docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md [hunk 2]
```diff
@@ -57,18 +61,20 @@ Edit `plugins/todo-list-backend/src/service/router.ts`:
...
 import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-node';
 import { todoListCreatePermission } from '@internal/plugin-todo-list-common';
 /* highlight-add-end */
 
 export interface RouterOptions {
-  logger: Logger;
+  logger: LoggerService;
+  /* highlight-add-next-line */
+  auth: AuthService;
   identity: IdentityApi;
```
Comment (freben) [inline docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md:78]: remove?

File: docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md [hunk 3]
```diff
@@ -134,30 +141,47 @@ export async function createRouter(
...
+          await createRouter({
+            logger,
+            /* highlight-add-start */
+            auth,
+            permissions,
+            /* highlight-add-end */
+            httpAuth,
+          }),
+        );
+        httpRouter.addAuthPolicy({
```
Comment (freben) [inline docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md:177]: is this part of the rest of the tutorial or can it be skipped to get less unrelated code?

Comment (camilaibs) [reply docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md:177]: This code is based on the previous setup step, which uses [this](https://github.com/backstage/backstage/blob/8f85ab3b04e3311171ca281d722c5177a5bba576/plugins/example-todo-list-backend/src/plugin.ts#L44) example plugin code as the base for the tutorial. I just used that as the base for the snippets here too.

**Review by freben — APPROVED** [review]
